zipkin-storage/cassandra-v1/src/main/java/zipkin2/storage/cassandra/v1/CassandraSpanConsumer.java

```diff
@@ -91,22 +90,22 @@
       spansToIndex.add(span);
     }
 
-    List<Call<ResultSet>> calls = new ArrayList<>();
+    List<Call<Void>> calls = new ArrayList<>();
     for (InsertTrace.Input insert : insertTraces) {
       calls.add(insertTrace.create(insert));
     }
     for (String insert : insertServiceNames) {
-      calls.add(insertServiceName.create(insert));
+      insertServiceName.maybeAdd(insert, calls);
     }
     for (InsertSpanName.Input insert : insertSpanNames) {
-      calls.add(insertSpanName.create(insert));
+      insertSpanName.maybeAdd(insert, calls);
     }
-    for (Map.Entry<String, String> entry : autocompleteTags) {
-      calls.add(insertAutocompleteValue.create(entry));
+    for (Map.Entry<String, String> autocompleteTag : autocompleteTags) {
+      insertAutocompleteValue.maybeAdd(autocompleteTag, calls);
     }
     indexer.index(spansToIndex.build(), calls);
-    if (calls.size() == 1) return calls.get(0).map(r -> null);
-    return new StoreSpansCall(calls);
+    if (calls.size() == 1) return calls.get(0);
+    return AggregateCall.newVoidCall(calls);
```

Comment:
this aggregate void call is used in each span consumer that doesnt do server side batching (or writes to multiple batches)